api/app/Http/Middleware/AuthMiddleware.php

```diff
@@ -0,0 +1,58 @@
+<?php namespace App\Http\Middleware;
+
+use Closure;
+use Tymon\JWTAuth\JWTAuth;
+use Illuminate\Http\Request;
+use App\Exceptions\ForbiddenException;
+
+class AuthMiddleware
+{
+    /**
+     * JWT Auth
+     *
+     * @var JWTAuth
+     */
+    protected $jwtAuth;
+
+    /**
+     * Assign dependencies.
+     *
+     * @param  JWTAuth  $jwtAuth
+     * @return void
+     */
+    public function __construct(JWTAuth $jwtAuth)
+    {
+        $this->jwtAuth = $jwtAuth;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  Request  $request
+     * @param  Closure  $next
+     * @return mixed
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        // Get possible x amount of parameters passed to the middleware
+        $userTypes = array_slice(func_get_args(), 2);
+
+        $token = $this->jwtAuth->getTokenFromRequest($request);
+        $user = $this->jwtAuth->getUser($token);
+
+        // Check the restriction types
+        if (in_array('admin', $userTypes) and $user->user_type == 'admin') {
```

Comment:
Yea, andre and I were looking at https://github.com/Zizaco/entrust or https://github.com/BeatSwitch/lock-laravel Can you take a look and them and give a recommendation on what we should do?